ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java

```diff
@@ -859,6 +859,12 @@ private void dispatchOverflowDraw(Canvas canvas) {
                     Math.max(bottomLeftBorderRadius - borderWidth.bottom, 0),
                   },
                   Path.Direction.CW);
+              // for sdk 24-28 fix issue with clipPath and LAYER_TYPE_NONE
```

Comment:
Why do we check it here but not in `setBorderRadius` for example?

Reply:
Thanks a lot @shergin 

I did not notice and I tried to implement it.

My impression is that the properties `overflow` and `borderRadius` from the `<View style={style} />` prop are sent through the react native bridge and saved with different `set` methods as instance variables in java. I believe the `borderRadius` should be drawn independently from the `overflow` property, then the `overflow` should be applied and crop the view.

https://github.com/facebook/react-native/blob/37e7b41419516d1d03cfd631d6bacef8fd2429a5/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java#L862

`setBorderRadius` could be called when `mOverflow` instance variable has value of `null`, the variable `mOverflow` could be later set to value `ViewProps.HIDDEN` **or** the `setBorderRadius` could be called after applying the [`crop` overflow hidden effect](https://github.com/facebook/react-native/blob/37e7b41419516d1d03cfd631d6bacef8fd2429a5/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java#L862) 

https://github.com/facebook/react-native/blob/aee88b6843cea63d6aa0b5879ad6ef9da4701846/ReactAndroid/src/main/java/com/facebook/react/uimanager/LayoutShadowNode.java#L568-L588

The logic inside `setBorderRadius` will disable `HARDWARE_ACCELLERATION` for device under [api 18](https://developer.android.com/reference/android/os/Build.VERSION_CODES) ([unsupported drawing operations](https://developer.android.com/guide/topics/graphics/hardware-accel#unsupported)).


https://github.com/facebook/react-native/blob/37e7b41419516d1d03cfd631d6bacef8fd2429a5/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java#L256-L264

As displayed below, my solution enables Hardware Accelleration for views with `mOverflow == ViewProps.HIDDEN` ([the white `view` with red borders](https://github.com/facebook/react-native/pull/28881/files#diff-f1a68e066841d3aef5c6a157a9cb4bceR279)) and with sdk 24-28, which has a relevant effect when applying animations on the view. The below low quality gifs images may not clearly display the difference. The below video titles are links to high quality videos hosted on my s3 space.

| **[ENABLE HARDWARE ACCELLERATION](https://fabriziobertoglio.s3.eu-central-1.amazonaws.com/opensource/react-native/fast_fps_animation.webm)** | **[DISABLE HARDWARE ACCELLERATION](https://fabriziobertoglio.s3.eu-central-1.amazonaws.com/opensource/react-native/slow_fps_animation.webm)** | 
|:-------------------------:|:-------------------------:|
| <img src="https://user-images.githubusercontent.com/24992535/82303149-d97c7300-99ba-11ea-9148-5d4a26fe0f59.gif" data-canonical-src="https://gyazo.com/eb5c5741b6a9a16c692170a41a49c858.png" width="200" height="" /> | <img src="https://user-images.githubusercontent.com/24992535/82302942-95896e00-99ba-11ea-9fc0-8822bddebecb.gif"  width="200" height="" /> |

Additional this is my `adb logcat` stack which displayes the `mOverflow` values from the above example. Every View instance from the above example will have a different value `mOverflow` depending on history (when the setter was called), for this reason I would avoid writing the overflow logic inside the method for another property (`setBorderRadius`).

I also want to add that I may be wrong and this is just my current opinion on this matter, I will add any improvement necessary to merge this pull request in master. Thanks a lot for you feedback. I remain available. I wish you a good day. Fabrizio

```
05-19 11:26:30.872  1854 15174 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.facebook.react.uiapp/.RNTesterActivity bnds=[440,1376][641,1698]} from uid 10031
05-19 11:26:30.920  1854  1871 I ActivityManager: Start proc 31855:com.facebook.react.uiapp/u0a67 for activity com.facebook.react.uiapp/.RNTesterActivity
05-19 11:26:31.676  1854  1877 I ActivityManager: Displayed com.facebook.react.uiapp/.RNTesterActivity: +774ms
05-19 11:26:32.530 31855 31893 I ReactNativeJS: Running "RNTesterApp" with {"rootTag":1}
05-19 11:26:32.602 31855 31894 W Test::  : called before LayoutShadowNode setOverflow(YogaOverflow.HIDDEN)
05-19 11:26:32.603 31855 31894 W Test::  : called after LayoutShadowNode setOverflow(YogaOverflow.HIDDEN)
05-19 11:26:32.611 31855 31894 W Test::  : called before LayoutShadowNode setOverflow(YogaOverflow.HIDDEN)
05-19 11:26:32.611 31855 31894 W Test::  : called after LayoutShadowNode setOverflow(YogaOverflow.HIDDEN)
05-19 11:26:32.631 31855 31855 W Test::  : setBorderRadius(float borderRadius) mOverflow: null
05-19 11:26:32.668 31855 31855 W Test::  : setBorderRadius(float borderRadius) mOverflow: null
05-19 11:26:32.689 31855 31855 W Test::  : dispatchOverflow mOverflow: hidden
05-19 11:26:32.698 31855 31855 W Test::  : dispatchOverflow mOverflow: hidden
05-19 11:26:32.703 31855 31894 W Test::  : called before LayoutShadowNode setOverflow(YogaOverflow.HIDDEN)
05-19 11:26:32.703 31855 31894 W Test::  : called after LayoutShadowNode setOverflow(YogaOverflow.HIDDEN)
05-19 11:26:32.722 31855 31894 W Test::  : called before LayoutShadowNode setOverflow(YogaOverflow.HIDDEN)
05-19 11:26:32.722 31855 31894 W Test::  : called after LayoutShadowNode setOverflow(YogaOverflow.HIDDEN)
05-19 11:26:32.732 31855 31894 W Test::  : called before LayoutShadowNode setOverflow(YogaOverflow.HIDDEN)
05-19 11:26:32.732 31855 31894 W Test::  : called after LayoutShadowNode setOverflow(YogaOverflow.HIDDEN)
05-19 11:26:32.814 31855 31855 W Test::  : setBorderRadius(float borderRadius, int position) mOverflow: null
05-19 11:26:32.814 31855 31855 W Test::  : setBorderRadius(float borderRadius, int position) mOverflow: null
05-19 11:26:32.900 31855 31855 W Test::  : setBorderRadius(float borderRadius) mOverflow: hidden
05-19 11:26:32.904 31855 31855 W Test::  : setBorderRadius(float borderRadius) mOverflow: null
05-19 11:26:32.926 31855 31855 W Test::  : setBorderRadius(float borderRadius) mOverflow: hidden
05-19 11:26:33.058 31855 31855 W Test::  : dispatchOverflow mOverflow: hidden
05-19 11:26:33.074 31855 31855 W Test::  : dispatchOverflow mOverflow: hidden
```

Reply:
@shergin CircleCI did not run on this branch, because it probably run on my fork fabriziobertoglio1987/react-native. I installed CircleCI on my fork as I wanted to see the test passing before publishing the pull request, but unluckily android/ios test that use the emulator require a CircleCI 30 USD subscription to run on more powerful instances... 

I could just add a commit and push again to trigger those CircleCI test, if is that the issue with this pull request. As you can see the CircleCI works on all the other Pull Requests, because I later disabled CircleCI Automation on my Fork.

If you see some weaknesses or issues in this implementation, I would be happy to receive your feedback as I am still learning ReactNative source and I want to reach full understanding of the entire Project.

Thanks and sorry.
I appreciate you work and feedback
I wish you a good day
Fabrizio Bertoglio

Reply:
code review changes applied with commit https://github.com/facebook/react-native/pull/28881/commits/e1fc40e0fbb9419de4ff9bd3eb5945f2f7a8c702 Sorry Shergin. I made a mistake. Thanks for the code review :pray: :peace_symbol: 

**<details><summary>CLICK TO OPEN TESTS RESULTS</summary>**
<p>

overflow hidden with transform rotate on a View with borderRadius does not cause issues

| BEFORE | AFTER |
|:-------------------------:|:-------------------------:|
| <img src="https://user-images.githubusercontent.com/24992535/109036573-08521300-76ca-11eb-9295-375ca12c936c.png" width="200" height="" /> | <img src="https://user-images.githubusercontent.com/24992535/109034866-5c5bf800-76c8-11eb-99b7-6945e98fe3f3.png" width="700" height="" /> |

animation test did not experience issues

| AFTER | 
|:-------------------------:|
| <video src="https://user-images.githubusercontent.com/24992535/109035327-d2f8f580-76c8-11eb-8473-83f79b1efb56.mp4" width="700" height="" /> |

</p>
</details>